quic/transport/stream.nim

```diff
@@ -21,20 +21,26 @@ method enter*(state: StreamState, stream: Stream) {.base.} =
 method leave*(state: StreamState) {.base.} =
   discard
 
-method read*(state: StreamState): Future[seq[byte]] {.base, async.} =
-  doAssert false # override this method
+method read*(
+    state: StreamState
+): Future[seq[byte]] {.base, async: (raises: [CancelledError, StreamError, QuicError]).} =
+  doAssert false, "override this method"
 
-method write*(state: StreamState, bytes: seq[byte]) {.base, async.} =
-  doAssert false # override this method
+method write*(
+    state: StreamState, bytes: seq[byte]
+) {.base, async: (raises: [CancelledError, StreamError]).} =
+  doAssert false, "override this method"
 
-method close*(state: StreamState) {.base, async.} =
-  doAssert false # override this method
+method close*(
+    state: StreamState
+) {.base, async: (raises: [CancelledError, QuicError]).} =
+  doAssert false, "override this method"
```

Comment:
actually, `QuicError` seems to be raised unnecessarily because of code like this 

```
proc shutdownStream*(connection: Ngtcp2Connection, streamId: int64) =
  let conn = connection.conn.valueOr:
    raise newException(Ngtcp2ConnectionClosed, "connection no longer exists")
````

code like this raises exception unnecessarily. if  `connection.conn` is not set, just return. in other words calling `close()` of anything closed  should have no effect. similarly start of anything that is already started should have no effect.